### ISSUE TITLE
validate_signature broken with recently released version of REXML (3.2.5)

### ISSUE THREAD

**Issue by maxfelsher** [onelogin/ruby-saml#577]
With REXML 3.2.5 (security release from this morning) and ruby-saml 1.12.0, calling `validate_signature` leads to an exception:

```
REXML::ParseException: Garbage component exists at the end: <]>: </p:Response[@ID=$id]/ds:Signature]>
/usr/local/rvm/gems/ruby-2.7.2/gems/rexml-3.2.5/lib/rexml/parsers/xpathparser.rb:28:in `parse'
/usr/local/rvm/gems/ruby-2.7.2/gems/rexml-3.2.5/lib/rexml/xpath_parser.rb:80:in `parse'
/usr/local/rvm/gems/ruby-2.7.2/gems/rexml-3.2.5/lib/rexml/xpath.rb:78:in `match'
/usr/local/rvm/gems/ruby-2.7.2/gems/ruby-saml-1.12.0/lib/onelogin/ruby-saml/response.rb:829:in `validate_signature'
```
It seems it doesn't like the `]` at the end of the XPath that ruby-saml is trying to use. Is that character necessary?

**Comment by seandilda** [comment]
I ran into the same issue a few minutes.

**Comment by maxfelsher** [comment]
It looks like the closing square bracket was introduced in https://github.com/onelogin/ruby-saml/commit/059abe4cfde3e4afbc703d6eadba7831fd6ea343 when `"/p:Response[@ID=$id]"` was changed to `"/p:Response/ds:Signature]"`. I think that the bracket is unnecessary but wasn't breaking anything until now.

**Comment by pitbulk** [comment]
Thanks for reporting this.  The typo was fixed and 1.12.1 released
